src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,10 +38,43 @@ public static void main(String[] args) throws IOException {
         LibSetup.setupLibraries();
         StartupParams params = new StartupParams(args);
 
+        checkUniqueInstance(params); //java 9
         checkJavaVersion(params);
         SwingUtilities.invokeLater(() -> new Main(params));
     }
 
+    private static void checkUniqueInstance(StartupParams params) throws IOException {
+        if (params.has(StartupParams.LaunchArg.NO_WARN)) return;
+
+        Path filePath = Paths.get("curr.pid");
+        long currentPid = ProcessHandle.current().pid();
+
+        if (Files.exists(filePath)) {
+            List<String> fileContent = new ArrayList<>(Files.readAllLines(filePath));
+
+            if (fileContent.isEmpty()) {
+                Files.writeString(filePath, "\n" + currentPid, StandardOpenOption.APPEND);
+            } else {
+                for (int i = 0; i < fileContent.size(); i++) {
+                    if (ProcessHandle.of(Long.parseLong(fileContent.get(i))).isPresent()) {
```

Comment:
If the file contains anything that isn't a number, this will throw an exception when parsing. If something like that occurs you should be ignoring it not throwing an exception

Additionally,  you should be checking if the process handle info matches that of the same process (in other words, it is also a darkbot instance running, not just any other random program). You can check this like:
ProcessHandle other = ProcessHandle.of(...);
if (other.isPresent() && other.info().command()...)

You can check that the command is the same (eg: java.exe), and the args match (eg: -jar darkbot.jar)